### ISSUE TITLE
Wait for previous action may work incorrectly when used in multiple events

### ISSUE THREAD

**Issue by dop2000** [Scirra/Construct-3-bugs#5514]
## Problem description

For example, when there are two or more AJAX requests running at the same time, "Wait for previous action" may trigger in the wrong event:
![image](https://user-images.githubusercontent.com/36763350/157181198-24d12676-766b-4da2-bd44-c942cacca506.png)

And here is the same project, but event #2 is moved to a separate event sheet:

![image](https://user-images.githubusercontent.com/36763350/157181387-9a0345f1-11f8-4808-8af7-a03cd9122302.png)

## Attach a .c3p

https://www.dropbox.com/s/4bogs595wsef9ol/AJAX_bug.c3p?dl=0

## Observed result

Inconsistent results every time, depending on the number of AJAX calls, file size, order of events, computer performance etc.

## Expected result

I believe, as the name suggests,  "Wait for previous action" should wait for the completion of the previous async action **in the same event**. It should not be triggered by any other actions in other events.

## More details

**Affected browsers/platforms:** all

## System details

<details><summary>View details</summary>
Platform information
Browser: Chrome
Browser version: 99.0.4844.51
Browser engine: Chromium
Context: browser
Operating system: Windows
Operating system version: 10
Device type: desktop
Device pixel ratio: 1.5
Logical CPU cores: 16
Approx. device memory: 8 GB
User agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.51 Safari/537.36
C3 release: r285.3 (stable)
Language setting: en-US

Local storage
Storage quota (approx): 275 gb
Storage usage (approx): 99 mb (0%)
Persistant storage: No

Browser support notes
This list contains missing features that are not required, but could improve performance or user experience if supported.

Nothing is missing. Everything is OK!
WebGL information
Version string: WebGL 2.0 (OpenGL ES 3.0 Chromium)
Numeric version: 2
Supports NPOT textures: yes
Supports GPU profiling: yes
Supports highp precision: yes
Vendor: Google Inc. (AMD)
Renderer: ANGLE (AMD, AMD Radeon(TM) Graphics Direct3D11 vs_5_0 ps_5_0, D3D11-27.20.11020.10001)
Major performance caveat: no
Maximum texture size: 16384
Point size range: 1 to 1024
Extensions:

EXT_color_buffer_float
EXT_color_buffer_half_float
EXT_disjoint_timer_query_webgl2
EXT_float_blend
EXT_texture_compression_bptc
EXT_texture_compression_rgtc
EXT_texture_filter_anisotropic
EXT_texture_norm16
KHR_parallel_shader_compile
OES_texture_float_linear
WEBGL_compressed_texture_s3tc
WEBGL_compressed_texture_s3tc_srgb
WEBGL_debug_renderer_info
WEBGL_debug_shaders
WEBGL_lose_context
WEBGL_multi_draw
OVR_multiview2
Audio information
System sample rate: 48000 Hz
Output channels: 2
Output interpretation: speakers
Supported decode formats:

WebM Opus (audio/webm; codecs=opus)
Ogg Opus (audio/ogg; codecs=opus)
WebM Vorbis (audio/webm; codecs=vorbis)
Ogg Vorbis (audio/ogg; codecs=vorbis)
MPEG-4 AAC (audio/mp4; codecs=mp4a.40.5)
MP3 (audio/mpeg)
FLAC (audio/flac)
PCM WAV (audio/wav; codecs=1)
Supported encode formats:

WebM Opus (audio/webm; codecs=opus)
Video information
Supported decode formats:

WebM AV1 (video/webm; codecs=av01.0.00M.08)
MP4 AV1 (video/mp4; codecs=av01.0.00M.08)
WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)
Ogg Theora (video/ogg; codecs=theora)
H.264 (video/mp4; codecs=avc1.42E01E)
Supported encode formats:

WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)

</details>

**Comment by AshleyScirra** [comment]
This is kind of a mistake in your events, and kind of a limitation in the design of the AJAX object.

The expression `AJAX.LastData` only returns a single value for the last succeeded request. So if you run two requests in parallel, as your events do, one request's result can overwrite another request's result. In other words what you really want to do is say "get the last data for tag 1" and "get the last data for tag 2", but all you can do is get the last data for the last completed request, which is insufficient when using those events.

The AJAX object doesn't do that because the last data could be very large, e.g. if you downloaded a large file, and so if it held on to every request's result it would basically leak memory (unless we added memory management actions to AJAX, which seems like overkill). In fact it should probably clear `LastData` after 'On completed' triggers, but that would now be a breaking change and break other use cases like this one.

Basically for parallel requests like this, you cannot use *Wait for previous actions to complete* - instead use the 'On completed' trigger to identify specific tags completing and read `LastData` in those triggers.
